### PR TITLE
🐛 Source Postgres: Improve date handling

### PR DESCRIPTION
**Pull request by edgao**
## What
For https://github.com/airbytehq/airbyte/issues/14590

* Handle BCE dates correctly:
  * CDC now supports BCE (previously would incorrectly read them as CE dates)
  * JDBC no longer shifts BCE dates forward by one year

## How
PR is stacked on @yurii-bidiuk 's https://github.com/airbytehq/airbyte/pull/14534.

This PR adds several _extremely_ hacky pieces of code, but I wasn't able to find more reasonable implementations. There are several problems:
* the `java.sql` date/time classes have poor support for BCE dates, and actually mangle the values if you try to convert them to `java.time` types
* `java.time` classes use a year 0, which doesn't match common usage
* debezium emits different classes depending on whether it's running in snapshot or incremental mode
  * I tested this by manually running a few syncs (reset -> sync (verify snapshot mode) -> insert records into source DB -> sync (verify incremental mode).

I've done my best to add comments explaining the behavior that we're trying to work around, but could easily be convinced that this is all terrible.

This PR also fixes the timestamptz test cases, which previously _worked_ but were very unclear. They now explicitly create `TIMESTAMP WITH TIME ZONE` values, which are easier to reason about.

### Followup work

We should extend the abstract datatype test to verify the incremental CDC case. Currently the AbstractPostgresDatatypeTest only verifies snapshot mode's behavior. I don't want to block this feature on figuring out that test though; will file a new issue to implement this later.

## Recommended reading order
1. AbstractPostgresSourceDatatypeTest
2. PostgresSourceOperations - this is the fix for JDBC's BCE dates being off by one year
3. AbstractJdbcCompatibleSourceOperations - a lot of utility methods for handling various date/time objects
4. DateTimeConverter - all of the interesting debezium-related code changes are in here
5. PostgresConverter, PostgresCdcProperties - just some basic housekeeping

## 🚨 User Impact 🚨
Nope, just bugfixes. I'll do a minor bump anyway since it's a user-visible change though.

### Community member or Airbyter

- [x] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [x] Secrets in the connector's spec are annotated with `airbyte_secret`
- [x] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [x] Connector's `README.md`
    - [x] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [x] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [x] Create a non-forked branch based on this PR and test the below items on it
- [x] Build is successful
- [x] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [x] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

### CODE REVIEW

**Comment by edgao** [comment]
@subodh1810 this should be ready for review - I found a few new bugs earlier today but was able to run some syncs just now.

File: airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresCdcProperties.java [hunk 1]
```diff
@@ -27,7 +28,7 @@ private static Properties commonProperties() {
     props.setProperty("connector.class", "io.debezium.connector.postgresql.PostgresConnector");
 
     props.setProperty("converters", "datetime");
-    props.setProperty("datetime.type", "io.airbyte.integrations.debezium.internals.PostgresConverter");
+    props.setProperty("datetime.type", PostgresConverter.class.getName());
```
Comment (subodh1810) [inline airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresCdcProperties.java:31]: Nice! I like this.

**Review by subodh1810 — APPROVED** [review]

**Review by yurii-bidiuk — APPROVED** [review]

**Pull request merged**